internal/resources/grafana/resource_data_source_config.go

```diff
@@ -0,0 +1,195 @@
+)
+
+func resourceDataSourceConfig() *schema.Resource {
+	return &schema.Resource{
+
+		Description: `
+* [Official documentation](https://grafana.com/docs/grafana/latest/datasources/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/developers/http_api/data_source/)
+
+The required arguments for this resource vary depending on the type of data
+source selected (via the 'type' argument).
+`,
+
+		CreateContext: UpdateDataSourceConfig,
+		UpdateContext: UpdateDataSourceConfig,
+		ReadContext:   ReadDataSourceConfig,
+		DeleteContext: DeleteDataSourceConfig,
+		SchemaVersion: 1,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"org_id": orgIDAttribute(),
+			"uid": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Default:     nil,
+				ForceNew:    true,
+				Description: "Unique identifier. If unset, this will be automatically generated.",
+			},
+			"http_headers": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "Custom HTTP headers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"json_data_encoded": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Serialized JSON string containing the json data. This attribute can be used to pass configuration options to the data source. To figure out what options a datasource has available, see its docs or inspect the network data when saving it from the Grafana UI. Note that keys in this map are usually camelCased.",
+				ValidateFunc: validation.StringIsJSON,
+				StateFunc: func(v interface{}) string {
+					json, _ := structure.NormalizeJsonString(v)
+					return json
+				},
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					if oldValue == "{}" && newValue == "" {
+						return true
+					}
+					return common.SuppressEquivalentJSONDiffs(k, oldValue, newValue, d)
+				},
+			},
+			"secure_json_data_encoded": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				Description:  "Serialized JSON string containing the secure json data. This attribute can be used to pass secure configuration options to the data source. To figure out what options a datasource has available, see its docs or inspect the network data when saving it from the Grafana UI. Note that keys in this map are usually camelCased.",
+				ValidateFunc: validation.StringIsJSON,
+				StateFunc: func(v interface{}) string {
+					json, _ := structure.NormalizeJsonString(v)
+					return json
+				},
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					if oldValue == "{}" && newValue == "" {
+						return true
+					}
+					return common.SuppressEquivalentJSONDiffs(k, oldValue, newValue, d)
+				},
+			},
+		},
+	}
```

Comment:
I assume these are exactly the same as in the resource_data_source.go file. They could be extracted to common vars